ide/coqide/session.ml

```diff
@@ -118,7 +118,9 @@ let create_script coqtop source_buffer =
 
 let misc () = CDebug.(get_flag misc)
 
-type action_stack = Gtk.text_mark list option
+type action_stack_entry = [ | `Mark of Gtk.text_mark | `Begin ]
```

Comment:
Don't use algebraic variants, they have a lot of issues. Either use the `option` type or remove the backquotes from these constructors.

Reply:
Got it.  The OCaml doc doesn't explain algebraic variants very well.  Or for that matter, type infererence.